var/spack/repos/builtin/packages/slate/package.py

```diff
@@ -72,28 +72,48 @@ def cmake_args(self):
             '-DSCALAPACK_LIBRARIES=%s'    % spec['scalapack'].libs.joined(';')
         ]
 
-    examples_src_dir = 'examples'
+    def keystore(self, key, value=None):
+        """Set and get key/value pairs"""
+        filestore = join_path(self.install_test_root, key + '_keystore.txt')
+        if value is None:
+            with open(filestore, 'r') as in_file:
+                return in_file.read().strip()
+        else:
+            with open(filestore, 'w') as out_file:
+                out_file.write('{0}\n'.format(value))
 
     @run_after('install')
     def cache_test_sources(self):
+        if self.spec.satisfies('@2020.10.00'):
+            return
         """Copy the example source files after the package is installed to an
         install test subdirectory for use during `spack test run`."""
-        if self.spec.satisfies('@master'):
-            self.cache_extra_test_sources([self.examples_src_dir])
+        self.cache_extra_test_sources(['examples'])
+        """The keystore() hack will be unnecessary once specs are properly
+        exposed to the stand-alone test framework."""
+        self.keystore('cmake', join_path(self.spec['cmake'].prefix.bin, 'cmake'))
+        self.keystore('mpi', self.spec['mpi'].prefix)
+        self.keystore('blaspp', self.spec['blaspp'].prefix)
+        self.keystore('lapackpp', self.spec['lapackpp'].prefix)
 
     def test(self):
-        if not self.spec.satisfies('@master') or '+mpi' not in self.spec:
-            print('Skipping: stand-alone tests only run on master with +mpi')
+        if self.spec.satisfies('@2020.10.00') or '+mpi' not in self.spec:
+            print('Skipping: stand-alone tests')
             return
 
-        test_dir = join_path(self.install_test_root, self.examples_src_dir)
-        test_bld_dir = join_path(test_dir, 'build')
-        with working_dir(test_bld_dir, create=True):
-            cmake('..')
+        test_dir = join_path(self.test_suite.current_test_cache_dir,
+                             'examples', 'build')
+        with working_dir(test_dir, create=True):
+            cmake_bin = self.keystore('cmake')
+            prefixes = ';'.join([self.keystore('blaspp'),
+                                 self.keystore('lapackpp'),
+                                 self.keystore('mpi'),
```

Comment:
(question) is the keystore used other than here? If this is the only place it is used, you should be able to refer for example to:

```
self.spec['mpi'].prefix
```

Reply:
OK, I do see now that `self.spec['mpi']` is available, but `self.spec['cmake']` is not.  How can I make cmake available in the spec?  I tried adding `depends_on('cmake', type='run')` with no luck.

Reply:
Interesting: adding the `run` dep type as you did should work. Would you be able to try reinstalling (`spack uninstall slate; spack install slate; ...`) and let me know if that persists?

Reply:
OK, `depends_on('cmake', type='run')` does now work once I reinstalled the slate package.  I'm not sure if it is appropriate to elevate cmake to a run dependency just so that the tests will run.  Would it be better to do this or to use the keystore to save the location of the cmake binary (like strumpack does)?

Reply:
IMO `cmake` is in fact a run dependency if you want to run tests after the package is installed, so I think it's appropriate to label it as a run dependency in this case (and that is my preference). If you disagree then I think we can leave it as is, although I'd want to think on how to phrase the associated comments to make it clearer to other Spack devs what the issue is in the future.

Reply:
OK, making the change. Thanks.